### ISSUE TITLE
model.export(export_dir='.')  does not export labels nor model.

### ISSUE THREAD

**Issue by amgsharma** [tensorflow/tensorflow#41545]
**System information**
- OSx
- TensorFlow installed from source
-  tf.__version__  ='2.4.0-dev20200719'

**Command used to run the converter or code if you’re using the Python API**
Running the code in [this notebook](https://colab.research.google.com/drive/1_LZxdWIfYFpZorpUkzgx_MPZzdzi8Hhy?usp=sharing), the export is fine. The issue arises when running the same code locally in a virtualenv with py36 and the same packages as per `!pip install git+https://github.com/tensorflow/examples.git#egg=tensorflow-examples[model_maker]`   

```
model.export(export_dir='.')
```

**The output from the converter invocation**

```
WARNING:tensorflow:From /Users/akiva111admin/code/security/venv_tf2/lib/python3.6/site-packages/tensorflow/python/training/tracking/tracking.py:111: Model.state_updates (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
WARNING:tensorflow:From /Users/akiva111admin/code/security/venv_tf2/lib/python3.6/site-packages/tensorflow/python/training/tracking/tracking.py:111: Model.state_updates (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
2020-07-19 22:21:59.233332: W tensorflow/python/util/util.cc:348] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
WARNING:tensorflow:From /Users/akiva111admin/code/security/venv_tf2/lib/python3.6/site-packages/tensorflow/python/training/tracking/tracking.py:111: Layer.updates (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
WARNING:tensorflow:From /Users/akiva111admin/code/security/venv_tf2/lib/python3.6/site-packages/tensorflow/python/training/tracking/tracking.py:111: Layer.updates (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.
Instructions for updating:
This property should not be used in TensorFlow 2.0, as updates are applied automatically.
2020-07-19 22:22:11.638306: I tensorflow/core/grappler/devices.cc:78] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA or ROCm support)
2020-07-19 22:22:11.638403: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2020-07-19 22:22:11.689811: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:816] Optimization results for grappler item: graph_to_optimize
2020-07-19 22:22:11.689845: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:818]   function_optimizer: Graph size after: 912 nodes (656), 920 edges (664), time = 32.473ms.
2020-07-19 22:22:11.689850: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:818]   function_optimizer: function_optimizer did nothing. time = 1.228ms.
2020-07-19 22:22:14.652608: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:315] Ignored output_format.
2020-07-19 22:22:14.652634: W tensorflow/compiler/mlir/lite/python/tf_tfl_flatbuffer_helpers.cc:318] Ignored drop_control_dependency.
```

**Also, please include a link to the saved model or GraphDef**

```
ipdb> model.summary()                                                                                                                                                                
Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
hub_keras_layer_v1v2 (HubKer (None, 1280)              3413024   
_________________________________________________________________
dropout (Dropout)            (None, 1280)              0         
_________________________________________________________________
dense (Dense)                (None, 2)                 2562      
=================================================================
Total params: 3,415,586
Trainable params: 2,562
Non-trainable params: 3,413,024
_________________________________________________________________

type(model)                                                                                                                                                                    
<class 'tensorflow_examples.lite.model_maker.core.task.image_classifier.ImageClassifier'>

model.hparams                                                                                                                                                                  
HParams(train_epochs=5, do_fine_tuning=False, batch_size=32, learning_rate=0.005, momentum=0.9, dropout_rate=0.2)

model.model_spec.name                                                                                                                                                          
'efficientnet_lite0'

model.num_classes                                                                                                                                                              
2

```

**Failure details**

Expected output as per the [example here](https://www.tensorflow.org/lite/tutorials/model_maker_image_classification#step_3_evaluate_the_customized_model) looks like: 
```INFO:tensorflow:Saving labels in ./labels.txt
INFO:tensorflow:Saving labels in ./labels.txt.
INFO:tensorflow:Assets written to: /tmp/tmp886peoe2/assets
INFO:tensorflow:Assets written to: /tmp/tmp886peoe2/assets```


**RNN conversion support**
If converting TF RNN to TFLite fused RNN ops, please prefix [RNN] in the title.

**Any other info / logs**

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by amahendrakar** [comment]
@amgsharma,
I do not have the access to view the Colab notebook you have linked. Could you please grant the required permissions or upload the script here. Thanks!

**Comment by amgsharma** [comment]
Its the colab notebook found [here](https://www.tensorflow.org/lite/tutorials/model_maker_image_classification):

**Comment by amahendrakar** [comment]
Was able to reproduce the issue, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/d7039fea469616765d3ba58d61191cdc/41545.ipynb). Thanks!

**Comment by natowi** [comment]
I have this problem, too. (saving labels.txt to the export_dir - don´t want to search for it in the tmp folder)
Do I understand this right (from the TF 2.3 label) that this issue will be solved with the next version?

**Comment by amgsharma** [comment]
@lintian06 @amahendrakar is there an update to this issue ? Thanks!

**Comment by natowi** [comment]
The commands changed and some parts of the documentation/tutorials still use the old commands. For more recent versions the solution is here:

> labels are embedded in tflite metadata.
> use `model.export(export_dir='.', with_metadata=False) instead of model.export(export_dir='.')` to get labels.txt
> Check the source code for why, https://github.com/tensorflow/examples/blob/master/tensorflow_examples/lite/model_maker/core/task/custom_model.py#L130-L177

https://github.com/tensorflow/tensorflow/issues/43463#issuecomment-697167439

I tested this with the Flower_Classification_with_TFLite_Model_Maker demo and it worked. https://github.com/tensorflow/tensorflow/issues/43463

**Comment by rringham** [comment]
> The syntax changed and some parts of the documentation/tutorials still use the old syntax.

Can this please be added to the documentation?

**Comment by sanatmpa1** [comment]
@rringham,

Can you point out the link for documentation, which needs to be modified? Thanks!

**Comment by sanatmpa1** [comment]
@amgsharma,

Can you take a look at this [comment](https://github.com/tensorflow/tensorflow/issues/41545#issuecomment-746653192) from @natowi and let us know if it helps in resolving your issue? Thanks!
